### PR TITLE
metadata: move models to Arsenal

### PR DESCRIPTION
**Pull request by jeremyds**
The models are the module BucketInfo, ObjectMD, WebsiteConfiguration and
ReplicationConfiguration.
These models are needed for the S3SOFS feature into the cdmiclient
repository. The S3 one does not export modules and shared ones are in
Arsenal so move these model modules to it.

# Pull request template

### Motivation and context

Why is this change required? What problem does it solve?

### Related issues

Please use the following link syntaxes #600 to reference issues in the
current repository

### Add tests to cover the changes

New tests added or existing tests modified to cover all changes

### Sign your work

In order to contribute to the project, you must sign your work
https://github.com/scality/Guidelines/blob/master/CONTRIBUTING.md#sign-your-work

Thank you again for contributing! We will try to test and integrate the change
as soon as we can.

### CODE REVIEW

File: tests/unit/bucket/WebsiteConfiguration.js [hunk 1]
```diff
@@ -2,7 +2,7 @@ const assert = require('assert');
 const {
     WebsiteConfiguration,
     RoutingRule,
-} = require('../../../lib/metadata/WebsiteConfiguration');
+} require('arsenal').models.websiteconfiguration;
```
Comment (RachedBenMustapha) [inline tests/unit/bucket/WebsiteConfiguration.js]: Is an `=` missing?

Comment (jeremyds) [inline tests/unit/bucket/WebsiteConfiguration.js]: done

**Comment by jeremyds** [comment]
@ironman-machine go

**Comment by jeremyds** [comment]
@ironman-machine go

**Comment by jeremyds** [comment]
So, the tests failure are caused by two things:

- because the models code / tests in S3 has been modified since the PR adding them to Arsenal has been merged (by me....),
- a test is broken since the delimiterTools is exported in Arsenal (guilty again).

These two problems are handled in two PRs, in Arsenal and in S3. I will rebase this branch and relaunch the tests once they will be merded.

**Comment by rahulreddy** [comment]
@ironman-machine

**Pull request merged**